src/emulator/auth/operations.ts

```diff
@@ -2543,6 +2556,123 @@ function parsePendingCredential(
   return { user, signInProvider };
 }
 
+function createTenant(
+  state: ProjectState,
+  reqBody: Schemas["GoogleCloudIdentitytoolkitAdminV2Tenant"]
+): Schemas["GoogleCloudIdentitytoolkitAdminV2Tenant"] {
+  assert(state instanceof AgentProjectState, "((Can only create tenant in agent project.))");
+
+  const formattedPhoneNumbers: { [key: string]: string } = {};
+  if (reqBody.testPhoneNumbers) {
+    Object.entries(reqBody.testPhoneNumbers).forEach(([phoneNumber, value]) => {
+      formattedPhoneNumbers[formatPhoneNumber(phoneNumber)] = value;
+    });
+  }
+
+  const tenant = {
+    displayName: reqBody.displayName,
+    allowPasswordSignup: reqBody.allowPasswordSignup,
+    enableEmailLinkSignin: reqBody.enableEmailLinkSignin,
+    enableAnonymousUser: reqBody.enableAnonymousUser,
+    disableAuth: reqBody.disableAuth,
+    testPhoneNumbers: formattedPhoneNumbers,
+    mfaConfig: reqBody.mfaConfig,
+    tenantId: "", // Placeholder until one is generated
+  };
+
+  return state.createTenant(tenant);
+}
+
+function formatPhoneNumber(phoneNumber: string) {
+  assert(
+    phoneNumber.length < PHONE_NUMBER_MAX_LENGTH,
+    "TOO_LONG ((The string supplied was too long to parse.))"
+  );
+
+  // Only check that the phone number is reasonably formatted, but does not
+  // fully verify that phone number is valid.
+  // TODO: support unicode, alpha characters and extensions?
```

Comment:
Consider explaining this regex a bit in the comment (high level, what is it matching)

Also, I don't think you need to do `new RegExp`; the literal `/.../g` itself is already an instance of RegExp that is created at compile time.

Reply:
Added a comment that makes the regex a bit more readable and removed the `new RegExp`, thanks!